### DISCUSSION TITLE
Is there a way to output Event Data from powershell output to .csv file from sigma rules?

### DISCUSSION THREAD

**Discussion by jv2k4ever** [WithSecureLabs/chainsaw#151] [Q&A] [answered]
Hey I love this tool, thanks for making it. I was wondering if it's possible to output the same columns found in the powershell output (particularly Event Data fields)  to a csv? Would be super useful and I'm not seeing a way how... thanks!

**Comment by alexkornitzer** [comment, marked as answer]
Hey @jv2k4ever, so the Sigma mapping file (assuming you are using `sigma-event-logs-all.yml`) is generic and thus will output all the event data into 'Event Data' column, this should also happen with CSV output.

![Screenshot 2023-12-01 at 13 49 15](https://github.com/WithSecureLabs/chainsaw/assets/2750747/1cad05ba-649d-478e-97ff-7e7f1ffd278e)

<img width="699" alt="Screenshot 2023-12-01 at 13 52 46" src="https://github.com/WithSecureLabs/chainsaw/assets/2750747/294c35c5-1f24-49b5-886d-dd2067cbbeb3">

You can tweak the mapping file to output the columns that are relevant to you by changing the `visible` flag from `false` to `true`.

![Screenshot 2023-12-01 at 13 50 25](https://github.com/WithSecureLabs/chainsaw/assets/2750747/2ae8a792-3945-4020-9b20-f57b0fa57004)

The reason that the `PowerShell Script` file has tailored columns is because that is generated by a Chainsaw rule and not a Sigma rule, where the important columns are defined within the rule.

![Screenshot 2023-12-01 at 13 54 18](https://github.com/WithSecureLabs/chainsaw/assets/2750747/0be84adf-1af6-42c5-a7c7-49273c98483e)

**Reply by jv2k4ever** [reply]
Awesome, thanks so much for responding, this is great. I actually figured out what I needed , when I exported the .csv the event data field in excel is just the following "---" in the cells (i mistakenly thought this was blank), I realized all I had to do was go into Home - Format - and set both "autofill row height" and "autofit column width" and the event data is available. Feel free to close this out : ]
